src/sentry/utils/snowflake.py

```diff
@@ -55,3 +94,55 @@ def snowflake_id_generation():
     ID_VALIDATOR.validate(snowflake_id)
 
     return snowflake_id
+
+
+def get_redis_cluster(redis_key: str):
+    return redis.clusters.get("default").get_local_client_for_key(redis_key)
+
+
+def store_snowflake_to_redis(redis_key: str, timestamp: int, used_region_sequence: int) -> None:
+    cluster = get_redis_cluster(redis_key)
+    used_region_sequences = cluster.get(str(timestamp))
+    if used_region_sequences:
+        used_region_sequences = json.loads(used_region_sequences)
+    else:
+        used_region_sequences = []
+    used_region_sequences.append(used_region_sequence)
+    cluster.setex(str(timestamp), int(_TTL.total_seconds()), json.dumps(used_region_sequences))
```

Comment:
Having region sequence counters automatically expire after 5 min is a great idea. Saves us having to do clean up book keeping.